client/block-builder/src/lib.rs

```diff
@@ -173,20 +175,29 @@ where
 		let estimated_header_size = header.encoded_size();
 
 		let mut api = api.runtime_api();
+		let version = api
+			.api_version::<dyn BlockBuilderApi<Block>>(parent_hash)?
+			.ok_or_else(|| Error::VersionInvalid("BlockBuilderApi".to_string()))?;
 
 		if record_proof.yes() {
 			api.record_proof();
 		}
 
-		api.initialize_block_with_context(
-			parent_hash,
-			ExecutionContext::BlockConstruction,
-			&header,
-		)?;
-
-		let version = api
-			.api_version::<dyn BlockBuilderApi<Block>>(parent_hash)?
-			.ok_or_else(|| Error::VersionInvalid("BlockBuilderApi".to_string()))?;
+		let executive_mode = if version >= 5 {
+			api.initialize_block_with_context(
+				parent_hash,
+				ExecutionContext::BlockConstruction,
+				&header,
+			)?
+		} else {
+			#[allow(deprecated)]
+			api.initialize_block_before_version_5_with_context(
+				parent_hash,
+				ExecutionContext::BlockConstruction,
+				&header,
```

Comment:
https://github.com/paritytech/substrate/blob/63246b699d7e2645c8b12aae46f8f0765c682183/client/transaction-pool/src/api.rs#L258 probably not that important anymore, but we also should do there the versioning.

Reply:
Okay calling `initialize_block_before_version_5` now since it is in a branch with `version < 3`.